scripts/check_frontend_coverage.py

```diff
@@ -221,6 +221,7 @@
     'logic-question-editor.directive.ts',
     'login-required-message.directive.ts',
     'maintenance-page.controller.ts',
+    'math-editor.directive.ts',
```

Comment:
Please add tests for this and `position-of-terms-editor`. Read the note on top, we discourage adding new files to this list.
/cc @marianazangrossi

Reply:
@kevintab95 I'm not sure exactly how we would test these directives. They don't have any functions that can be unit tested, they just display an input box/dropdown which takes in values and stores them in a variable. Seems like that behavior would be tested by the e2e tests, right?

Reply:
In `math-editor`, I see a bunch of variables being initialised and an event listener being added. You can check if this method works as expected in the test (ie. it initialises correctly). Ditto about `position-of-terms-editor`. @rohitkatlaa recently worked on adding tests for directives/components he had worked on as part of his GSoC project, you can ask him if you are stuck.

Also, in general, it is good practice to write unit tests for any code that you write. e2e tests cannot be used as a replacement for unit tests. Note that, as mentioned at the top of this list, this should not be updated unless you get a confirmation from @marianazangrossi.

Reply:
Thanks for the explanation, @kevintab95! I've added tests for the `position-of-terms-editor` component. While adding tests for the `math-editor` component, I'm running into an issue that I can't seem to be able to resolve.
In the test file, I've called the `$onInit` method of the component which initializes a guppy instance. When I ran the tests, it threw an error: `ReferenceError: Guppy is not defined`. This is most likely due to the way guppy has been imported. As per instructions given [here](https://guppy.js.org/site/doc/quickstart), I've imported the lib's files in guppy.html directly with the script tag.
Also, apparently, _guppy-js lists no main file and has no index.js, so it can't be directly required._. ([ref](https://npm.runkit.com/guppy-js)). Please let me know if you have any idea on how to resolve this, thanks!

/cc @aks681 @rohitkatlaa @marianazangrossi

Reply:
Can you simply mock Guppy instead?

Reply:
Done!